File: Client/DriverHub.Client/src/app/core/services/reservation/models/reservation-response.ts
export enum ReservationStatus {
  Pending = 1,
  Approved = 2, // Backend's existing Confirmed value.
  Cancelled = 3,
  Completed = 4
}

export const reservationStatusLabels: Record<ReservationStatus, string> = {
  [ReservationStatus.Pending]: 'Onay Bekliyor',
  [ReservationStatus.Approved]: 'Onaylandı',
  [ReservationStatus.Cancelled]: 'İptal Edildi',
  [ReservationStatus.Completed]: 'Tamamlandı'
};

export interface ReservationResponse {
  id: string;
  carId: string;
  brandName: string;
  model: string;
  plate: string;
  customerFirstName: string | null;
  customerLastName: string | null;
  customerEmail: string | null;
  customerPhone: string | null;
  pickupLocationId: string;
  pickupLocationName: string;
  returnLocationId: string;
  returnLocationName: string;
  startDate: string;
  endDate: string;
  createdDate: string;
  status: ReservationStatus;
  processedAt: string | null;
  processedByName: string | null;
  rentalDays: number;
  basePrice: number;
  extraPrice: number;
  insurancePrice: number;
  totalPrice: number;
}

File: Core/DriverHub.Application/Features/Entities/Reservations/Common/ReservationResponse.cs
using DriverHub.Domain.Enums;

namespace DriverHub.Application.Features.Entities.Reservations.Common;

public sealed record ReservationResponse(
    Guid Id, Guid CarId, string BrandName, string Model, string Plate,
    string? CustomerFirstName, string? CustomerLastName, string? CustomerEmail, string? CustomerPhone,
    Guid PickupLocationId, string PickupLocationName, Guid ReturnLocationId, string ReturnLocationName,
    DateTime StartDate, DateTime EndDate, DateTime CreatedDate, ReservationStatus Status,
    DateTime? ProcessedAt, string? ProcessedByName,
    decimal BasePrice, decimal ExtraPrice, decimal InsurancePrice, decimal TotalPrice)
{
    public int RentalDays => ReservationTimePolicy.GetRentalDays(StartDate, EndDate);
}
